### ISSUE TITLE
State of selected dates not announced on VoiceOver via keyboard navigation

### ISSUE THREAD

**Issue by pliu-coursera** [airbnb/react-dates#2195]
**react-dates version**
react-dates@20.2.5

**Describe the bug**
We are using `DayPickerSingleDateController` in our project and the following behaviour violates our accessibility standards:

When a user turns on VoiceOver and navigates through the calendar dates using keyboard, the state of the selected dates needs be announced on change. However, currently, if users press enter to select a new date, the selected state is not announced and users would only be informed that the date is selected when they navigate away from that selected date and navigate back. This results in users who rely heavily on screen readers to lose information on the current state of the calendar.

**Desktop:**
 - OS: Windows 10, Mac OS Catalina 10.15
 - Browser: Chrome, Firefox, Safari
